parla/task_runtime.py

```diff
@@ -1179,18 +1173,14 @@ def _update_resource(self, dev: Device, res: str, amount: float, block: bool):
                 dres = self._devices[dev]
                 if -amount <= dres[res]:
                     dres[res] += amount
-                    if amount > 0:
-                        self._monitor.notify_all()
-                    assert dres[res] <= dev.resources[res], "{}.{} was over deallocated".format(dev, res)
-                    assert dres[res] >= 0, "{}.{} was over allocated".format(dev, res)
+                    # TODO(lhc): Due to floating point, it doesn't work.
+                    #assert dres[res] <= dev.resources[res], "{}.{} was over deallocated".format(dev, res)
+                    #assert dres[res] >= 0, "{}.{} was over allocated".format(dev, res)
```

Comment:
Gross. Can we make these integers?

Reply:
one possible way is use the default vcu sum as the number of threads or something bigger numbers, not 1.
@dialecticDolt  is it fine?

Reply:
Also gross. We might add the resources back up and exceed the original amount bc of rounding errors and the order that tasks complete matters too. 
Probably using: https://docs.python.org/3/library/fractions.html ? might be the cleanest way to handle this.